src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -135,12 +132,39 @@ public class NuGetPackageManager
 
             InitializePackagesFolderInfo(PackagesFolderPathUtility.GetPackagesFolderPath(SolutionManager, Settings), excludeVersion);
             DeleteOnRestartManager = deleteOnRestartManager;
+        }
 
-            string globalPackagesFolder = BuildIntegratedProjectUtility.GetEffectiveGlobalPackagesFolderOrNull(solutionManager.SolutionDirectory, settings);
-            if (globalPackagesFolder != null)
+        /// <summary>
+        /// SourceRepositories for the user global package folder and all fallback package folders.
+        /// </summary>
+        public IReadOnlyList<SourceRepository> GlobalPackageFolderRepositories
+        {
+            get
             {
-                var globalPackagesSource = new Configuration.PackageSource(globalPackagesFolder);
-                GlobalPackagesFolderSourceRepository = SourceRepositoryProvider.CreateRepository(globalPackagesSource, FeedType.FileSystemV3);
+                if (_globalPackageFolderRepositories == null)
+                {
+                    var sources = new List<SourceRepository>();
+
+                    // Read package folders from settings
+                    var pathContext = NuGetPathContext.Create(Settings);
+                    var folders = new List<string>();
+                    folders.Add(pathContext.UserPackageFolder);
+                    folders.AddRange(pathContext.FallbackPackageFolders);
+
+                    foreach (var folder in folders)
+                    {
+                        // Create a repo for each folder
+                        var source = SourceRepositoryProvider.CreateRepository(
+                            new PackageSource(folder),
+                            FeedType.FileSystemV3);
+
+                        sources.Add(source);
+                    }
```

Comment:
This will normally be 1, sometimes 2. I'm not sure this is going to have any impact considering this is done once per creation of the package manager.